### PR TITLE
add python solution for mooyo mooyo (860)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-860.mdx [hunk 1]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
 
 </JavaSection>
 
+<PySection>
+```python
+import sys
+from collections import deque
+
+sys.stdin = open("mooyomooyo.in", "r")
+sys.stdout = open("mooyomooyo.out", "w")
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:264]: .

File: solutions/usaco-860.mdx [hunk 2]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
...
+
+input = sys.stdin.readline
+
+# possible directions to visit all connected regions
+directions = [
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:269]: rename to DIRECTIONS & make it a tuple since it's constant

File: solutions/usaco-860.mdx [hunk 3]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
...
+			# boundaries
+			if 0 <= nx < 10 and 0 <= ny < a:
+				# check if already visited
+				if graph[ny][nx] == graph[y][x] and not visited[ny][nx]:
+					q.appendleft((nx, ny))
+					visited[ny][nx] = True
+					coors.append((nx, ny))
+	return coors
+
+def mooyo(graph):
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:315]: doesn't have to be in functions

File: solutions/usaco-860.mdx [hunk 4]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
...
+	(-1, 0), 
+	(0, -1)
+]
+
+a, b = map(int, input().split())
+
+def gravity(graph, s, coors):
+
+	for i in range(s // 2):
+		for j in range(5):
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:281]: magic number

File: solutions/usaco-860.mdx [hunk 5]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
...
+		graph[y].append(0)
+    
+	for i in range(5):
+		for j in range(s // 2):
+			graph[i][j], graph[j][i] = graph[j][i], graph[i][j]
+    
+	return graph
+
+# bfs to mark all connected regions
+def bfs(x, y, graph, visited):
```
Comment (SansPapyrus683) [inline solutions/usaco-860.mdx:295]: does this need to be in a function as well?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-860.mdx [hunk 6]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
...
+	
+	q = deque()
+	q.append((x, y))
+        visited[y][x] = True
+        coors = [(x, y)]
```
Comment (ryanchou-dev) [inline solutions/usaco-860.mdx:300]: maybe `coords` instead?

File: solutions/usaco-860.mdx [hunk 7]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
 
 </JavaSection>
 
+<PySection>
+```python
```
Comment (ryanchou-dev) [inline solutions/usaco-860.mdx:259]:
```suggestion
```py
```
doesn't really matter ^^

File: solutions/usaco-860.mdx [hunk 8]
```diff
@@ -255,4 +255,108 @@ public class MooyoMooyo {
...
+
+		y = a - 1
+		while blocks:
+			node = blocks.pop()
+			graph[y][x] = node
+			y -= 1
+    
+	return mooyo(graph)
+    
+
```
Comment (ryanchou-dev) [inline solutions/usaco-860.mdx:349]:
```suggestion
```

**Review by ryanchou-dev — COMMENTED** [review]
i don't think BFS is intended \ add it as an alternate sol instead?

**Comment by jjliewie** [comment]
> i don't think BFS is intended \ add it as an alternate sol instead?

wait really? i thought this problem was a classic bfs (or dfs i just prefer bfs) problem

**Comment by ryanchou-dev** [comment]
> > i don't think BFS is intended \ add it as an alternate sol instead?
> 
> wait really? i thought this problem was a classic bfs (or dfs i just prefer bfs) problem

yeah it's a DFS prob (curr BFS is a gold concept)

actually, i think it's fine, just add a comment saying "this can also be implemented with DFS" or smtg

**Comment by jjliewie** [comment]
> > > i don't think BFS is intended \ add it as an alternate sol instead?
> > 
> > 
> > wait really? i thought this problem was a classic bfs (or dfs i just prefer bfs) problem
> 
> yeah it's a DFS prob (curr BFS is a gold concept)
> 
> actually, i think it's fine, just add a comment saying "this can also be implemented with DFS" or smtg

yeah okay. i mean u can interchange between dfs or bfs & i personally think bfs should be in silver lol. why is it a gold concept?

File: solutions/usaco-860.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx:315]: ill take it out

File: solutions/usaco-860.mdx (hunk 4 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx:281]: yep okay

File: solutions/usaco-860.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx:318]: nvm can't, its recursive

File: solutions/usaco-860.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx]: done

File: solutions/usaco-860.mdx (hunk 4 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx]: changed

File: solutions/usaco-860.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx:298]: yes. kinda.

File: solutions/usaco-860.mdx (hunk 6 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx:303]: okay ill change it

File: solutions/usaco-860.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/usaco-860.mdx:301]: ok then give a better name than `bfs` and also add type annotations (other functions too)

**Comment by ryanchou-dev** [comment]
> > > > i don't think BFS is intended \ add it as an alternate sol instead?
> > > 
> > > 
> > > wait really? i thought this problem was a classic bfs (or dfs i just prefer bfs) problem
> > 
> > 
> > yeah it's a DFS prob (curr BFS is a gold concept)
> > actually, i think it's fine, just add a comment saying "this can also be implemented with DFS" or smtg
> 
> yeah okay. i mean u can interchange between dfs or bfs & i personally think bfs should be in silver lol. why is it a gold concept?

difficulty inflation 😖

**Review by ryanchou-dev — APPROVED** [review]
⭐

**Review by envyaims — APPROVED** [review]
more comments especially in the mooyo function like the ones in other languages pls but overall seems fine

File: solutions/usaco-860.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-860.mdx:301]: what should i name it

Comment (SansPapyrus683) [reply solutions/usaco-860.mdx:301]: `mark_connected` or something

**Comment by SansPapyrus683** [comment]
show me ac screen and i'll merge

**Comment by jjliewie** [comment]
> show me ac screen and i'll merge
Screen Shot 2022-06-12 at 5 29 20 AM

**Pull request merged**